### ISSUE TITLE
Parser return type

### ISSUE THREAD

**Issue by Ko-Khan** [nus-cs2103-AY2324S2/forum#180]
What should the parser class return? Could I get it to interact with storage or tasklist directly? Or should it return a string instead?

**Comment by Jerome-j** [comment]
I was inspired by the AB2 project so what I did was the parser takes in the raw user input, then there is a regex to split up the command, and route to the respective function. The logic to parse the command, and the logic to handle the parsed command is handled differently. Perhaps you could visit https://github.com/se-edu/addressbook-level2 to be inspired.

I think you can directly interface with Storage or TaskList, but this might pose some challenge when your app needs to scale horizontally to add in more features - as the Java Class file might grow too big to manage.

**Comment by zhekaiii** [comment]
Hi, for me I kinda followed the example they gave. I implemented the Command (abstract) class and my parse method uses regex/split to retrieve the relevant information and returns the relevant Command.

**Comment by tsulim** [comment]
Similar to @Jerome-j, I've also taken inspiration from [AB2 project](https://github.com/se-edu/addressbook-level2). What I did was to have my parser return the respective Command class along with the relevant information/arguments.

You could get it to interact with storage or tasklist directly but I wouldn't recommend it as I think that the Parser purpose is to mainly parse user's input and direct them to where they are needed to be.

**Comment by sebin0817** [comment]
My approach is similar to Jerome's, regex was used to split up the command before each part was routed to my parser. More specifically, my parser focused on accepting String to return LocalDateTime.

**Comment by Ko-Khan** [comment]
Thanks for the help. I will refrain from using my parser to interact with the other classes

**Comment by Arixeyeion** [comment]
For me I treated it as Parser is the class that decides what commands should be called as it is the one the processes all the info. I split the input string and then fed it into different commands based on the different scenarios, calling other classes like tasklist, todo/mark/unmark/etc and their methods where applicable like tl.mark(integer of desired task). I also called ui to display messages whenever needed. While I do think what the others have suggested is better in the sense of OOP, that's how I have implemented my Parser.
